contracts/src/v0.8/ccip/pools/SiloedLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,181 @@
+    if (s_siloedChainSelectors[releaseOrMintIn.remoteChainSelector]) {
+      s_lockedTokensByChainSelector[releaseOrMintIn.remoteChainSelector] -= localAmount;
+    }
+
+    // Release to the recipient
+    getToken().safeTransfer(releaseOrMintIn.receiver, localAmount);
+
+    emit Released(msg.sender, releaseOrMintIn.receiver, localAmount);
+
+    return Pool.ReleaseOrMintOutV1({destinationAmount: localAmount});
+  }
+
+  /// @notice Returns whether the tokens locked for a given remote chain should be siloed independently
+  /// from all other remote chains.
+  /// @param remoteChainSelector the CCIP specific selector for the remote chain being interacted with.
+  /// @return isSiloed Whether the funds should be isolated from all the others.
+  function chainFundsAreSiloed(
+    uint64 remoteChainSelector
+  ) external view returns (bool isSiloed) {
+    return s_siloedChainSelectors[remoteChainSelector];
+  }
+
+  /// @notice Returns the amount of tokens in the token pool that were locked for a specific chain selector.
+  /// @param remoteChainSelector the CCIP specific selector for the remote chain being interacted with.
+  /// @return lockedTokens The tokens locked into this token pool for the given selector, zero if the remote chain
+  /// funds are not siloed.
+  function getLockedTokensByChain(
+    uint64 remoteChainSelector
+  ) external view returns (uint256 lockedTokens) {
+    return s_lockedTokensByChainSelector[remoteChainSelector];
+  }
+
+  /// @notice Updates designations for chains on whether to mark funds as Siloed or not
+  /// @param removes A list of chain selectors to disable Siloeing
+  /// @param adds A list of chain selectors to enable LR instead of BM. These chains must not have been migrated
+  /// to CCTP yet or the transaction will revert
+  function updateChainSelectorMechanisms(uint64[] calldata removes, uint64[] calldata adds) external onlyOwner {
+    // If removing, change designation and update tokens as being available for all chains to use.
+    for (uint256 i = 0; i < removes.length; ++i) {
+      delete s_siloedChainSelectors[removes[i]];
+      delete s_lockedTokensByChainSelector[removes[i]];
+
+      emit ChainSiloeDesignationUpdated(removes[i], false);
+    }
+
+    for (uint256 i = 0; i < adds.length; ++i) {
+      s_siloedChainSelectors[adds[i]] = true;
+      emit ChainSiloeDesignationUpdated(adds[i], true);
+    }
+  }
+
+  /// @notice Gets the rebalancer able to provide liquidity for a remote chain selector
+  /// @param remoteChainSelector The CCIP specific selector for the remote chain being interacted with.
+  /// @return The current liquidity manager, owner if the chain's funds are not siloed.
+  function getRebalancerByChain(
+    uint64 remoteChainSelector
+  ) external view returns (address) {
+    if (s_siloedChainSelectors[remoteChainSelector]) return s_rebalancerByChain[remoteChainSelector];
```

Comment:
Why do rebalancers only work for siloed chains? That breaks the current operations we do for weth